### ISSUE TITLE
Expose cli options to construct offline mirror

### ISSUE THREAD

**Issue by olafurpg** [scalacenter/scalafix#106]
It should be possible to run semantic rewrites from the cli with `--sourcepath` and `--classpath` options. Those options are necessary to construct an offline scala.meta.Mirror, see here: https://github.com/scalameta/scalameta/blob/baf4bc3c3b77d692750c7aec89c6ff66b1f1cf9e/scalahost/src/main/scala/scala/meta/Mirror.scala#L12

**Comment by stuhood** [comment]
I'll give this a shot today.

**Comment by olafurpg** [comment]
Cool! Heads up, we just split scalahost into a library part and compiler plugin part https://github.com/scalameta/scalameta/pull/775 Feel free to bump up the scalameta/scalahost dependency in scalafix to 

```scala
resolvers += Resolver.bintrayIvyRepo("scalameta", "maven")
libraryDependencies += "org.scalameta" %% "scalameta" % "1.7.0-485-086c4ac9"
```

I'm OK with depending on pre-release versions of scalameta on master branch in the scalafix repo .

The offline Mirror constructor we want to call from scalafix-cli is this one here: https://github.com/scalameta/scalameta/blob/086c4ac9ff63da2278d40f80b4e389d70fc3f1a6/scalahost/core/src/main/scala/scala/meta/Mirror.scala#L39 Please let me know if you feel something is missing from the docstring.

**Comment by olafurpg** [comment]
BTW, please ignore the weird formatting of parameters in ScalafixOptions, it's been a bug in scalafmt which I've been meaning to fix for a while but never gotten around to.

**Comment by stuhood** [comment]
There is a draft of this available on #107.

**Comment by olafurpg** [comment]
Awesome! Thank you for this contribution.
